File: d671.11716-Digital_Fortress/main.py
def isint(num):
    if num-int(num) == 0:
        return True
    return False


row = input()
for i in range(int(row)):
    string = input()
    ans = ""
    if isint(len(string)**0.5):
        for k in range(int(len(string)**0.5)):
            j = k
            while j < len(string):
                ans += string[j]
                j += int(len(string)**0.5)
        print(ans, '\n', sep="")

    else:
        print("INVALID\n")

File: d671.11716-Digital_Fortress/main.cpp
#include<iostream>
#include<math.h>
#include<string>
using namespace std;
bool isint(double num)
{
	if (num - (int)num == 0)return true;
	return false;
}
int main(void)
{
	int row,size;
	string str,ans;
	cin >> row;
	cin.ignore();
	for (int i = 0; i < row; i++)
	{
		ans = "";
		str = "";
		size = 0;
		
		getline(cin, str);
		if (isint(sqrt(str.size())))
		{
			size = sqrt(str.size());
			for (int k = 0; k < size; k++)
			{
				for (int j = k; j < size*size; j += size)
				{
					ans += str[j];
				}
			}
			cout << ans<<'\n';
		}
		else cout << "INVALID\n";
	}
	return 0;
}